### PR TITLE
Multilingualism

### PR DESCRIPTION
**Pull request by tigercat2000**
This PR adds the ability to speak in multiple languages throughout the same message. You are able to freely transition between any language you know as if you were typing seperate messages, simply by adding the language's shortcut key into the middle of the message. 

Nonverbal/signlang flags have been removed from saycode as well due to difficulty supporting them with the new system - In practice, this means nothing, as those flags were not used. Noise language (the emotes you can use over the radio) still functions fine, though it did require some significant code snowflaking to baby it's fragile ass back to functioning. 

Telecomms seems to work just fine with the changes from my testing, but it's possible that it has issues. NTSL definitely works, but other stuff that relies on the signal system such as signallers or airlock controllers may have suffered - I didn't get a chance to test those tonight. 

On the more technical side:
This constitutes a full overhaul of how language parsing works in saycode, and is *very* likely to break stuff as a result. 
Messages are now split up into a list() of /datum fragments when they are parsed for languages - there are some helper functions to adapt legacy code to this system without having to rewrite it, but most systems have already been converted over to take advantage of the fragment system. One advantage this bestows upon us is that we don't have to fuck around with returns so much anymore, as was the case with the horrific messes like handle_speech_problem - because we can just modify the datum itself. This may cause issues down the line if I missed some expectation somewhere that say messages aren't mutable, but I didn't see any issues from my ~8h of local testing. 
As a random fun note about this: The say verb and a few other miscellaneous behaviors are determined based on the first language detected. 
My code is absolutely a thrown-together mess - I'd appreciate suggestions on how to improve both it's performance and it's styling, as it's giving me a very heavy "tree" kinda feeling. 

Things I've tested:
 - Mob speech
   - Humans
   - Xenos
   - Borgs
   - AIs
   - Drones
   - Renault
 - Holopads <holocalls & AI>
 - Multilingual speech
 - Normal Radios, extensively
 - NTTC
 - Mech radio
 - Megaphones
 - Recorders
 - Status effects
 - The horse head
 - Muzzles
 - AI/Ghost say tracking hrefs
 - universal_speak
 - Binary radios

Things I didn't:
 - Signallers
 - Airlocks
 - Grenade/voice activated assemblies in general
 - Hivemind languages (partial - I made sure they worked in _triggering_ them, but then tweaked some speech spans, I didn't get to check if they have any issues)
 - universal_understand (Should work the same as universal_speak but I didn't test it)
 - Some mobs
   - Cortical Borers
   - Brains/MMIs/Posibrains
   - Poly
   - SNPCS
   - Slimes
   - pAIs
 - Hivemind radios
 - Robotic slurring
 - Whispering with anything but humans
 - Videocams
 - Spacepods/Mecha cargo contents being able to hear inside 
 - Nested say (like a pAI in a pAI shell in a bag in a mob)

:cl:
add: You can now speak in multiple languages in the same message! This works for all forms of speech, including radios. Minus megaphones because fuck em.
fix: Some mild formatting errors and inconsistencies that have historically been present within mob speech.
fix: Holopads behaving a bit weirdly with how they interact with the typical say system
/:cl:

### CODE REVIEW

**Comment by Spartan6** [comment]
As long as it doesn’t add French.

**Comment by Necaladun** [comment]
Completely approve the theory of this - there's no reason you shouldn't be able to mix and match languages in one sentence.

Final approval though is totally up to maints as to how much it fucks with saycode etc. I could see it being ugly!

**Comment by PhantasmicDream** [comment]
The ability to swear in another language with out having to break up the sentence, I'm excited!

**Comment by Citinited** [comment]
Finally we can swear in Chinese while speaking Common for everything else!

**Comment by Kyep** [comment]
I suspect that this might be related to the language changes:

[2018-11-24T17:41:09] Runtime in eye.dm,146: undefined proc or verb /mob/living/carbon/human/relay speech(). 
   proc name: hear say (/mob/camera/aiEye/hear_say)
   usr: Customs Officer Ian (CKEY) (/mob/living/simple_animal/pet/corgi/Ian)
   usr.loc: The carpet (123,130,1) (/turf/simulated/floor/carpet)
   src: Camera Eye (CHARNAME) (/mob/camera/aiEye/remote)
   src.loc: the floor (124,126,1) (/turf/simulated/floor/wood)
   call stack:
   Camera Eye (CHARNAME) (/mob/camera/aiEye/remote): hear say("waf", "barks", null, 0, Customs Officer Ian (/mob/living/simple_animal/pet/corgi/Ian), null, null)
   Customs Officer Ian (/mob/living/simple_animal/pet/corgi/Ian): say("waf", null, "barks", 1, 0, 0)
   Customs Officer Ian (/mob/living/simple_animal/pet/corgi/Ian): Say("waf")
   Customs Officer Ian (/mob/living/simple_animal/pet/corgi/Ian): .Say()

File: code/modules/mob/say.dm [hunk 1]
```diff
@@ -132,17 +132,77 @@
...
+	speaking = new_speaking
+	if(new_message)
+		message = new_message
+
+/mob/proc/find_valid_prefixes(message)
+	var/list/prefixes = list() // [["Common", start, end], ["Gutter", start, end]]
+	for(var/i in 1 to length(message))
+		var/selection = trim_right(lowertext(copytext(message, i, i + 3)))
+		var/datum/language/L = GLOB.language_keys[selection]
+		if(L != null && can_speak_language(L)) // What the fuck... remove the L != null check if you ever find out what the fuck is adding `null` to the languages list on absolutely random mobs... seriously what the hell...
```
Comment (KasparoVy) [inline code/modules/mob/say.dm]: Lmfao, fucking magic

**Comment by KasparoVy** [comment]
Wow, that's hardcore. You beast

This will be awesome! Though, how the hell will gesture based languages ever work with this system if we ever implement those?

**Comment by Spartan6** [comment]
The whisper verb under IC does not work.

File: code/modules/mob/living/carbon/human/whisper.dm [hunk 2]
```diff
@@ -1,13 +0,0 @@
-//Lallander was here
```
Comment (KasparoVy) [inline code/modules/mob/living/carbon/human/whisper.dm]: There goes myyy heeroooo

**Pull request merged**